test/src/filebrowser/crumbs.spec.ts

```diff
@@ -0,0 +1,223 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+
+import expect = require('expect.js');
+
+import {
+  ServiceManager
+} from '@jupyterlab/services';
+
+import {
+  Message, sendMessage
+} from 'phosphor/lib/core/messaging';
+
+import {
+  Widget, WidgetMessage
+} from 'phosphor/lib/ui/widget';
+
+import {
+  simulate
+} from 'simulate-event';
+
+import {
+  BreadCrumbs, FileBrowserModel
+} from '../../../lib/filebrowser';
+
+
+
+const ITEM_CLASS = 'jp-BreadCrumbs-item';
+
+
+class LogCrumbs extends BreadCrumbs {
+  methods: string[] = [];
+  events: string[] = [];
+
+  handleEvent(event: Event): void {
+    super.handleEvent(event);
+    this.events.push(event.type);
+  }
+
+  protected onAfterAttach(msg: Message): void {
+    super.onAfterAttach(msg);
+    this.methods.push('onAfterAttach');
+  }
+
+  protected onBeforeDetach(msg: Message): void {
+    super.onBeforeDetach(msg);
+    this.methods.push('onBeforeDetach');
+  }
+
+  protected onUpdateRequest(msg: Message): void {
+    super.onUpdateRequest(msg);
+    this.methods.push('onUpdateRequest');
+  }
+}
+
+
+describe('filebrowser/model', () => {
+
+  let manager: ServiceManager.IManager;
+  let model: FileBrowserModel;
+  let crumbs: LogCrumbs;
+  let first: string;
+  let second: string;
+  let third: string;
+  let path: string;
+
+  before((done) => {
+    manager = new ServiceManager();
+    let contents = manager.contents;
+    contents.newUntitled({ type: 'directory' }).then(cModel => {
+      first = cModel.name;
+      return contents.newUntitled({ path: cModel.path, type: 'directory' });
+    }).then(cModel => {
+      second = cModel.name;
+      return contents.newUntitled({ path: cModel.path, type: 'directory' });
+    }).then(cModel => {
+      third = cModel.name;
+      path = cModel.path;
+    }).then(done, done);
+  });
+
+  beforeEach((done) => {
+    model = new FileBrowserModel({ manager });
+    model.cd(path).then(() => {
+      crumbs = new LogCrumbs({ model });
+    }).then(done, done);
+  });
+
+  afterEach(() => {
+    model.dispose();
+  });
+
+  describe('BreadCrumbs', () => {
+
+    describe('#constructor()', () => {
+
+      it('should create a new BreadCrumbs instance', () => {
+        let bread = new BreadCrumbs({ model });
+        expect(bread).to.be.a(BreadCrumbs);
+        let items = crumbs.node.getElementsByClassName(ITEM_CLASS);
+        expect(items.length).to.be(1);
+      });
+
+      it('should add the jp-BreadCrumbs class', () => {
+        expect(crumbs.hasClass('jp-BreadCrumbs')).to.be(true);
+      });
+
+    });
+
+    describe('#handleEvent()', () => {
+
+      context('click', () => {
+
+        it('should switch to the parent directory', (done) => {
+          Widget.attach(crumbs, document.body);
```

Comment:
`Widget.attach(crumbs, document.body` is being called in multiple tests, but `crumbs` is not being detached. Is this intentional?